datascience_eda/datascience_eda.py

```diff
@@ -41,31 +95,71 @@ def get_clustering_default_hyperparameters():
     Examples
     -------
     >>> hyper_dict = get_clustering_default_hyperparameters()
-    >>> hyper_dict["distance_metric"] = "cosine"
-    >>> hyper_dict["K-Means"]["n_clusters"] = (1, 10)
-    >>> hyper_dict["DBSCAN"]["eps"] = [1]
+    >>> hyper_dict["KMeans"]["n_clusters"] = range(2, 10)
+    >>> hyper_dict["DBSCAN"]["eps"] = [0.3]
     >>> hyper_dict["DBSCAN"]["min_samples"] = [3]
-    >>> initial_clustering(X, hyperparameter_dict=hyper_dict)
+    >>> hyper_dict["DBSCAN"]["distance_metric"] = "cosine"
+    >>> explore_clustering(X, hyperparameter_dict=hyper_dict)
     """
     clustering_default_hyperparameters = {
-        "distance_metric": "euclidean",
-        "K-Means": {"n_clusters": (3, 10)},
-        "DBSCAN": {"eps": (1, 3), "min_samples": (3, 10)},
+        "KMeans": {"n_clusters": range(2, 9)},
+        "DBSCAN": {
+            "eps": [0.5],
+            "min_samples": [5],
+            "distance_metric": "euclidean",
+        },
     }
     return clustering_default_hyperparameters
 
 
+def plot_pca_clusters(data, labels):
+    """Carries out dimensionality reduction on the data for visualization, apdated from Lecture 2
+
+    Parameters
+    ----------
+    data : pandas.DataFrame
+        the dataset
+    labels : list
+        list of labels predicted
+
+    Returns
+    -------
+    matplotlib.axes.AxesSubPlot
+        the PCA plot
+    """
+    pca = PCA(n_components=2)
+    principal_comp = pca.fit_transform(data)
+    pca_df = pd.DataFrame(
+        data=principal_comp, columns=["pca1", "pca2"], index=data.index
+    )
+    pca_df["cluster"] = labels
+    # fig = plt.figure(figsize=(6, 4))
+    fig = sns.scatterplot(
+        x="pca1", y="pca2", hue="cluster", data=pca_df, palette="tab10"
+    )
+    plt.show()
+    plt.clf()
+    plt.close()
+
+    return fig
+
+
+# endregion
+
+# region clustering functions
 def explore_clustering(
-    df, hyperparameter_dict=get_clustering_default_hyperparameters()
+    df, numeric_cols=None, hyperparameter_dict=get_clustering_default_hyperparameters()
 ):
     """fit and plot K-Means, DBScan clustering algorithm on the dataset
 
     Parameters
     ----------
     df : pandas.DataFrame
-        the dataset (X)
+        the dataset (X), should already transformed with StandardScaler
+    numeric_cols: list, optional
+        a list of numeric columns used for clustering, by default None, will be assigned with all numeric columns
     hyperparameter_dict : dict, optional
-        the distance metric and hyperparameters to be used in the clustering algorithm, by default get_clustering_default_hyperparameters()
+        the hyperparameters to be used in the clustering algorithms, by default get_clustering_default_hyperparameters()
```

Comment:
How would it be if the values of hyper parameters are directly mentioned in place of/after function name so that the user need not go and check get_clustering_default_hyperamaeters() function?